src/Tasks/LargeFileUploadTask.php

```diff
@@ -0,0 +1,345 @@
+<?php
+namespace Microsoft\Graph\Core\Tasks;
+
+use DateTime;
+use DateTimeInterface;
+use Exception;
+use GuzzleHttp\Psr7\Utils;
+use Http\Promise\Promise;
+use Http\Promise\RejectedPromise;
+use InvalidArgumentException;
+use Microsoft\Graph\Core\Errors\LargeFileUploadTaskErrors\Error404GetUploadStatusException;
+use Microsoft\Graph\Core\Errors\LargeFileUploadTaskErrors\Error405GetUploadStatusException;
+use Microsoft\Graph\Core\Models\LargeFileTaskUploadSession;
+use Microsoft\Graph\Core\Models\LargeFileUploadCreateUploadSessionBody;
+use Microsoft\Kiota\Abstractions\HttpMethod;
+use Microsoft\Kiota\Abstractions\RequestAdapter;
+use Microsoft\Kiota\Abstractions\RequestInformation;
+use Microsoft\Kiota\Abstractions\Serialization\AdditionalDataHolder;
+use Microsoft\Kiota\Abstractions\Serialization\Parsable;
+use Psr\Http\Message\StreamInterface;
+use RuntimeException;
+use SplQueue;
+
+class LargeFileUploadTask
+{
+    /** @var Parsable|LargeFileTaskUploadSession */
+    private $uploadSession;
+    private RequestAdapter $adapter;
+    private StreamInterface $stream;
+    private int $uploadedChunks = 0;
+    private int $chunks;
+    private ?string $nextRange = null;
+    private int $fileSize;
+    private int $maxChunkSize;
+    private int $uploaded = 0;
+    public function __construct(Parsable $uploadSession, RequestAdapter $adapter, StreamInterface $stream, int $maxChunkSize = 5 * 1024 * 1024){
+        $this->uploadSession = $uploadSession;
+        $this->adapter = $adapter;
+        $this->stream = $stream;
+        $this->fileSize = $stream->getSize();
+        $this->maxChunkSize = $maxChunkSize;
```

Comment:
I left this out since different APIs provide different constraints on the size. 
For example.
Outlook provides different limits for attachment Item.
https://learn.microsoft.com/en-us/graph/outlook-large-attachments?tabs=http

Reply:
Would it make sense to default max chunk size to the smaller of the two? 4MB for Outlook?

Reply:
Done